app/misc/metric_builder.rb

```diff
@@ -1,16 +1,30 @@
 class MetricBuilder
   def run_all(content_item_attributes)
-    metrics = all_metrics.map do |metric_klass|
-      metric_klass.new(content_item_attributes).run
-    end
-    metrics.reduce(:merge)
+    run(all_metrics, content_item_attributes)
+  end
+
+  def run_collection(content_items)
+    run(collection_metrics, content_items)
   end
 
 private
 
+  def run(metrics_to_run, data)
+    metrics = metrics_to_run.map do |metric_klass|
+      metric_klass.new(data).run
+    end
+    metrics.reduce(:merge)
+  end
+
   def all_metrics
```

Comment:
If we are adding the method `collection_metrics`, I would rename `all_metrics` to `content_item_metrics`? The only difference seems to be what is the target of the calculation.

As a side note, it would be good to remove the suffix in both metrics, as we have the class scoped. I think this is my bad, as I started using the suffix myself.

Feel free to add separate commits for these two things, if you are happy with my feedback.